modules/renter/contractor/downloader.go

```diff
@@ -79,7 +80,12 @@ func (hd *hostDownloader) Sector(root crypto.Hash) ([]byte, error) {
 
 	hd.contractor.mu.Lock()
 	hd.contractor.contracts[contract.ID] = contract
-	hd.contractor.saveSync()
+	cpath := fmt.Sprintf("contracts.%s", contract.ID.String())
+	hd.contractor.persist.update(
+		cpath+".lastrevision", contract.LastRevision,
+		cpath+".lastrevisiontxn", contract.LastRevisionTxn,
+		cpath+".downloadspending", contract.DownloadSpending,
+	)
```

Comment:
The standard way to make that workable I think would be to add getters and setters to every field, which is awful naively, but if you grouped fields well maybe it would not be so bad. Of course, the list of getters and setters would only grow with time.